src/compiler/parser.ts

```diff
@@ -8617,8 +8617,19 @@ namespace Parser {
 
         // Parses out a JSDoc type expression.
         export function parseJSDocTypeExpression(mayOmitBraces?: boolean): JSDocTypeExpression {
-            const pos = getNodePos();
+            let pos = getNodePos();
             const hasBrace = (mayOmitBraces ? parseOptional : parseExpected)(SyntaxKind.OpenBraceToken);
+            if (token() === SyntaxKind.AtToken) {
+                    // skip @link since it's a common mistake

```

Comment:
I'm assuming this skips anything, not just `@link`?

Reply:
Yes, it's parse failure have `@throws {@anything type}` -- you'll just see a different parse error than before because the `@` will be skipped. I should probably add a test to show that off, since it will apply to checked JS files too.

Reply:
Yeah, I just saw the comment above and it didn't line up with the condition, so wanted to make sure it was intentional.